pkg/apis/kubevirt/v1alpha1/types_cloudprofile.go

```diff
@@ -45,3 +49,22 @@ type MachineImageVersion struct {
 	// SourceURL is the url of the image
 	SourceURL string `json:"sourceURL"`
 }
+
+// MachineTypes extend machineType object to KubeVirt provider specific config
+type MachineTypes struct {
```

Comment:
The name of this type should be `MachineType`, as it refers to a single machine type. `MachineImages` is similarly wrong and could be renamed to `MachineImage` as well.

Reply:
`MachineType` is incorrect I agree, but when it comes to `MachineImages` it's correct.
`MachineImages` structure can contain more than one image (list of versions with source URL).

Although it requires some refactoring, the part about getting machine images has been copy-pasted from the vsphere provider, I think it can be simplified.  Currently we have two `MachineImages(s)` structures, in worker and cloud profile files, we could just use one. I am going to create an issue for that were I explain the problem and how to solve that.

Reply:
OK, agreed. Let's discuss the `MachineImages` simplification in person before opening issues or making changes.